crates/hdk/tests/integration.rs

```diff
@@ -0,0 +1,570 @@
+#![cfg(feature = "mock")]
+use std::ops::Range;
+
+use hdk::prelude::*;
+use holo_hash::hash_type::AnyLinkable;
+use Compare::*;
+
+mod integrity_zomes {
+    use holochain_deterministic_integrity::prelude::*;
+
+    #[hdk_entry_helper]
+    pub struct A;
+    #[hdk_entry_helper]
+    pub struct B;
+    #[hdk_entry_helper]
+    pub struct C;
+
+    pub mod integrity_a {
+
+        use super::*;
+
+        #[hdk_entry_defs(skip_hdk_extern = true)]
+        #[unit_enum(UnitFoo)]
+        pub enum EntryTypes {
+            A(A),
+            B(B),
+            C(C),
+        }
+
+        #[hdk_link_types]
+        pub enum LinkTypes {
+            A,
+            B,
+            C,
+        }
+    }
+
+    pub mod integrity_b {
+        use super::*;
+
+        #[hdk_entry_defs(skip_hdk_extern = true)]
+        #[unit_enum(UnitFoo)]
+        pub enum EntryTypes {
+            A(A),
+            B(B),
+            C(C),
+        }
+
+        #[hdk_link_types]
+        pub enum LinkTypes {
+            A,
+            B,
+            C,
+        }
+    }
+}
+
+#[hdk_entry_zomes]
```

Comment:
I wonder if there could be a slightly more descriptive name for this indicating that these are the types found in the dependencies? 

- `hdk_entry_type_dependencies` / `hdk_link_type_dependencies`? 
- `hdk_entry_type_deps` / `hdk_link_type_deps`?
- `hdk_dependent_entry_types` / `hdk_dependent_link_types`?

Reply:
I like the last one